sdk/monitor/azure-opentelemetry-exporter-azuremonitor/src/test/java/com/azure/opentelemetry/exporter/azuremonitor/AppConfigurationExporterIntegrationTest.java

```diff
@@ -0,0 +1,70 @@
+package com.azure.opentelemetry.exporter.azuremonitor;
+
+import com.azure.core.util.FluxUtil;
+import com.azure.data.appconfiguration.ConfigurationClient;
+import com.azure.data.appconfiguration.ConfigurationClientBuilder;
+import io.opentelemetry.api.trace.Span;
+import io.opentelemetry.api.trace.Tracer;
+import io.opentelemetry.context.Scope;
+import org.junit.jupiter.api.Tag;
+import org.junit.jupiter.api.Test;
+import reactor.core.publisher.Mono;
+
+import java.nio.charset.StandardCharsets;
+import java.util.Optional;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+@Tag("integration")
+public class AppConfigurationExporterIntegrationTest extends AzureMonitorExporterTestBase {
+
+    @Test
+    public void setConfigurationTest() throws InterruptedException {
+        CountDownLatch appConfigCountDown = new CountDownLatch(1);
+        CountDownLatch exporterCountDown = new CountDownLatch(2);
+
+        ConfigurationClient client = getConfigurationClient(appConfigCountDown);
+        Tracer tracer = configureAzureMonitorExporter((context, next) -> {
+            Mono<String> asyncString = FluxUtil.collectBytesInByteBufferStream(context.getHttpRequest().getBody())
```

Comment:
Watch out on FluxUtil consuming the request body to completion. `collectByesInByteBufferStream` will mutate the underlying `ByteBuffer`s' `remaining()` value which could cause `hasRemaining()` to be false when the networking layer attempts to send the request